loclist.kak

```diff
@@ -0,0 +1,147 @@
+                }
+            }
+            set-option buffer grep_current_line %val{cursor_line}
+            evaluate-commands -try-client %opt{jumpclient} -verbatim -- edit -existing -- %reg{a} %reg{b} %reg{c}
+            try %{ focus %opt{jumpclient} }
+        }
+    }
+}
+
+define-command loclist-next-location -docstring %{
+    loclist-next-location
+    Jump to next location listed in the current location list buffer, usually one of
+    *diagnostics* *goto* *grep* *implementations* *lint-output* *make* *references* *symbols*
+
+    %opt{loclist_buffer} determines the buffer current location list buffer.
+    %opt{loclist_location_format} determines matching locations.
+} -buffer-completion %{
+    evaluate-commands -try-client %opt{jumpclient} -save-regs / %{
+        buffer %opt{loclist_buffer}
+        set-register / %opt{loclist_location_format}
+        execute-keys ge %opt{grep_current_line}g<a-l> /<ret>
+        loclist-jump
+    }
+    try %{
+        evaluate-commands -client %opt{toolsclient} %{
+            buffer %opt{loclist_buffer}
+            execute-keys gg %opt{grep_current_line}g
+        }
+    }
+}
+
+define-command loclist-previous-location -docstring %{
+    loclist-previous-location
+    Jump to previous location listed in the current location list buffer, usually one of
+    *diagnostics* *goto* *grep* *implementations* *lint-output* *make* *references* *symbols*
+
+    %opt{loclist_buffer} determines the buffer current location list buffer.
+    %opt{loclist_location_format} determines matching locations.
+} -buffer-completion %{
+    evaluate-commands -try-client %opt{jumpclient} -save-regs / %{
+        buffer %opt{loclist_buffer}
+        set-register / %opt{loclist_location_format}
+        execute-keys ge %opt{grep_current_line}g<a-h> <a-/><ret>
+        loclist-jump
+    }
+    try %{
+        evaluate-commands -client %opt{toolsclient} %{
+            buffer %opt{loclist_buffer}
+            execute-keys gg %opt{grep_current_line}g
+        }
+    }
+}
+
+define-command -override loclist-stack-push -docstring "record location list buffer" %{
+    evaluate-commands %sh{
+        eval set -- $kak_quoted_opt_loclist_stack
+        if printf '%s\n' "$@" | grep -Fxq -- "$kak_bufname"; then {
+            exit
+        } fi
+        newbuf=$kak_bufname-$#
+        echo "try %{ delete-buffer! $newbuf }"
+        echo "rename-buffer $newbuf"
+        echo "set-option -add global loclist_stack %val{bufname}"
+    }
+    set-option global loclist_buffer %val{bufname}
+}
+
+define-command -override loclist-stack-pop -docstring "restore location list buffer" %{
+    evaluate-commands %sh{
+        eval set -- $kak_quoted_opt_loclist_stack
+        if [ $# -eq 0 ]; then {
+            echo fail "loclist-stack-pop: no location list buffer to pop"
+            exit
+        } fi
+        printf 'set-option global loclist_stack'
+        top=
+        while [ $# -ge 2 ]; do {
```

Comment:
For the record, the brace grouping in my config exists to enable my autoformatter, and for large blocks enable the `m` command.
It's not necessary here, so probably better to drop it;
it could be confusing to readers because we don't need an extra level of command grouping.